### PR TITLE
Add custom de/serialization for CoinType

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add custom de/serialization for CoinType, so the bindings can provide the number for it

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

Added test

## Change checklist

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/account/types/mod.rs [hunk 1]
```diff
@@ -228,3 +230,58 @@ impl TryFrom<u32> for CoinType {
...
+// Custom serialization to return the coin type number
+impl Serialize for CoinType {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        match *self {
+            CoinType::IOTA => serializer.serialize_u32(self.clone() as u32),
+            CoinType::Shimmer => serializer.serialize_u32(self.clone() as u32),
+        }
```
Comment (Tuditi) [inline src/account/types/mod.rs]: Why use a match statement if the actions undertaken are the same for both options?

**Review by Tuditi — APPROVED** [review]
Looks good to me! I had a minor question out of interest.

File: src/account/types/mod.rs (hunk 1 above)
Comment (Thoralf-M) [reply src/account/types/mod.rs]: Good point, you're right shouldn't be a match there
Initially I provided the matching coin type number hardcoded, so there it wasn't the same, but didn't remove the match after changing it, will do now

File: src/account/types/mod.rs [hunk 2]
```diff
@@ -204,10 +205,11 @@ impl From<u32> for AccountIdentifier {
 }
 
 /// BIP 44 coin type
-#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
```
Comment (thibault-martinez) [inline src/account/types/mod.rs]: Make it copy since it's backed by an u32, will avoid some explicit clones

File: src/account/types/mod.rs [hunk 3]
```diff
@@ -228,3 +230,55 @@ impl TryFrom<u32> for CoinType {
...
     }
 }
+
+// Custom serialization to return the coin type number
+impl Serialize for CoinType {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_u32(self.clone() as u32)
```
Comment (thibault-martinez) [inline src/account/types/mod.rs]: Like this

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**